src/app/src/test/kotlin/org/imperial/mrc/hint/unit/controllers/ADRControllerTests.kt

```diff
@@ -752,11 +752,78 @@ class ADRControllerTests : HintrControllerTests()
     }
 
     @Test
-    fun `creates a release`()
+    fun `creates a release if no release exists on ADR with the same name`()
     {
-        val expectedUrl = "dataset_version_create"
+        val data = mapOf("data" to listOf<Any>())
         val mockClient = mock<ADRClient> {
-            on { post(expectedUrl, listOf("dataset_id" to "dataset-1", "name" to "release-1")) } doReturn ResponseEntity
+            on { get("/dataset_version_list?dataset_id=dataset-1") } doReturn ResponseEntity
+                    .ok()
+                    .body(objectMapper.writeValueAsString(data))
+            on { post("/dataset_version_create", listOf("dataset_id" to "dataset-1", "name" to "release-1")) } doReturn ResponseEntity
+                    .ok()
+                    .body("whatever")
+        }
+        val mockBuilder = mock<ADRClientBuilder> {
+            on { build() } doReturn mockClient
+        }
+        val sut = ADRController(
+                mock(),
+                mock(),
+                mockBuilder,
+                objectMapper,
+                mockProperties,
+                mock(),
+                mock(),
+                mockSession,
+                mock())
+        val result = sut.createRelease("dataset-1", "release-1")
+        assertThat(result.body!!).isEqualTo("whatever")
+    }
+
+    @Test
+    fun `deletes release of same name on ADR and creates a new release`()
+    {
+        val existingRelease = mapOf("name" to "release-1", "id" to "other-id")
+        val data = mapOf("data" to listOf(existingRelease))
+        val mockClient = mock<ADRClient> {
+            on { get("/dataset_version_list?dataset_id=dataset-1") } doReturn ResponseEntity
+                    .ok()
+                    .body(objectMapper.writeValueAsString(data))
+            on { post("/dataset_version_create", listOf("dataset_id" to "dataset-1", "name" to "release-1")) } doReturn ResponseEntity
+                    .ok()
+                    .body("whatever")
+            on { post("/version_delete", listOf("version_id" to "other-id")) } doReturn ResponseEntity
+                    .ok()
+                    .body("whatever")
+        }
+        val mockBuilder = mock<ADRClientBuilder> {
+            on { build() } doReturn mockClient
+        }
+        val sut = ADRController(
+                mock(),
+                mock(),
+                mockBuilder,
+                objectMapper,
+                mockProperties,
+                mock(),
+                mock(),
+                mockSession,
+                mock())
+        val result = sut.createRelease("dataset-1", "release-1")
+        assertThat(result.body!!).isEqualTo("whatever")
```

Comment:
Both endpoints are returning `"whatever"` so looks to me as if this test isn't really checking that `/version_delete` is being invoked. You can either change one of responses to a different string, or check that the mock function is actually called.